lib/deck.rb

```diff
@@ -0,0 +1,38 @@
+class Deck
+  attr_reader :cards, :high_ranked_cards
+
+  def initialize(cards = [])
+    @cards = cards
+    @high_ranked_cards = []
+  end
+  
+  def rank_of_card_at(index_position)
+    cards[index_position].rank
+  end
+  
+  def high_ranking_cards
```

Comment:
I noticed that you added an instance variable called `@high_ranked_cards` to store your high ranking cards. With the way the method is written this isn't necessary.

```ruby
def high_ranking_cards
  high_ranked_cards = [] #<== Using a local variable
  cards.each do |card|
    if card.rank >= 11
      high_ranked_cards << card #<== Adding cards to that local variable
    end
  end
  return high_ranked_cards #<== Returning that local variable at the end
end
```

We can take it a step further and use an even better enumerable for the situation, `#select` or `#find_all`, they are aliases.

```ruby
def high_ranking_cards
  cards.select do |card|
    card.rank >= 11
  end
end
```

The `select/find_all` enumerable only returns the elements that meet the condition. (It does not modify the original array)